Formula/zip.rb

```diff
@@ -0,0 +1,49 @@
+class Zip < Formula
+  desc "Compression and file packaging/archive utility"
+  homepage "http://www.info-zip.org/Zip.html"
+  url "https://downloads.sourceforge.net/project/infozip/Zip%203.x%20%28latest%29/3.0/zip30.tar.gz"
+  sha256 "f0e8bb1f9b7eb0b01285495a2699df3a4b766784c1765a8f1aeedf63c0806369"
+
+  keg_only :provided_by_osx
+
+  # Upstream is unmaintained so we use the Debian patchset:
+  # https://packages.debian.org/sid/zip
+  patch do
+    url "https://mirrors.ocf.berkeley.edu/debian/pool/main/z/zip/zip_3.0-11.debian.tar.xz"
+    mirror "https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/z/zip/zip_3.0-11.debian.tar.xz"
+    sha256 "c5c0714a88592f9e02146bfe4a8d26cd9bd97e8d33b1efc8b37784997caa40ed"
+    apply %w[
+      patches/01-typo-it-is-transferring-not-transfering
+      patches/02-typo-it-is-privileges-not-priviliges
+      patches/03-manpages-in-section-1-not-in-section-1l
+      patches/04-do-not-set-unwanted-cflags
+      patches/05-typo-it-is-preceding-not-preceeding
+      patches/06-stack-markings-to-avoid-executable-stack
+      patches/07-fclose-in-file-not-fclose-x
+      patches/08-hardening-build-fix-1
+      patches/09-hardening-build-fix-2
+      patches/10-remove-build-date
+    ]
+  end
+
+  def install
+    system "make", "-f", "unix/Makefile",
+      "CC=#{ENV.cc}",
+      "generic",
+      "BINDIR=#{bin}",
+      "MANDIR=#{man1}",
+      "install"
+  end
+
+  test do
+    (testpath/"test1").write "Hello!"
+    (testpath/"test2").write "Bonjour!"
+    (testpath/"test3").write "Moien!"
+
+    system "#{bin}/zip", "test.zip", "test1", "test2", "test3"
+    %w[test1 test2 test3].each do |f|
+      rm f
```

Comment:
I don't think I understand the test. If you're `rm`'ing the files, then of course they won't exist. And what does that have to do with the zipping above?

Reply:
Yeah, this part of the `unzip` test shouldn't have been lifted. Might want to check `test.zip` exists, which is perhaps where this was going & then forgotten? Unsure.

Reply:
@DomT4 is right. I started by adapting the `unzip` test for `zip`, and got distracted and forgot to remove that part.
I'll just check if the file exists instead.

Reply:
seems like we can just use the test from unzip …

Reply:
I would prefer the current shorter version of the test, so that I have nothing to change on Linuxbrew, because there unzip may not be installed, and I can't pull it in for the test.

But if you prefer the full test, I can add it back.

Reply:
Let's use `-T` on the zip file.